[Error] GeometryContainer compilation Clang 7.1.0

Compiling the current master branch with Clang 7.1.0 throws the next error 
`/home/rzorrilla/Kratos/kratos/containers/geometry_container.h:355:16: error: call to non-static member function without an object argument
        Flags::operator =(rOther);`

@tteschemacher it seems to be related to your last changes.

--- a/kratos/containers/geometry_container.h
+++ b/kratos/containers/geometry_container.h
@@ -24,7 +24,6 @@
 // Project includes
 #include "includes/define.h"
 #include "containers/pointer_hash_map_set.h"
-#include "containers/data_value_container.h"
 
 
 namespace Kratos
@@ -352,7 +351,6 @@ private:
     /// Assignment operator.
     GeometryContainer& operator=(const GeometryContainer& rOther)
     {
-        Flags::operator =(rOther);
         mGeometries = rOther.mGeometries;
     }
 

--- a/kratos/containers/geometry_container.h
+++ b/kratos/containers/geometry_container.h
@@ -352,6 +352,7 @@ private:
     GeometryContainer& operator=(const GeometryContainer& rOther)
     {
         mGeometries = rOther.mGeometries;
+        return *this;
     }
 
     ///@}

